llvm/test/CodeGen/AIE/aie2/GlobalISel/prelegalizercombiner-shufflevector.mir

```diff
@@ -682,3 +682,172 @@ body:             |
     %3:_(<128 x s8>) = G_SHUFFLE_VECTOR %1:_(<128 x s8>), %2:_(<128 x s8>), shufflemask(128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63)
     PseudoRET implicit $lr, implicit %3
 ...
+
+---
+name:            concat_vector_reverse_32_512
+legalized:       false
+body:             |
+  bb.1.entry:
+    liveins: $wl2, $wl4
+    ; CHECK-LABEL: name: concat_vector_reverse_32_512
+    ; CHECK: liveins: $wl2, $wl4
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(<8 x s32>) = COPY $wl2
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(<8 x s32>) = COPY $wl4
+    ; CHECK-NEXT: [[CONCAT_VECTORS:%[0-9]+]]:_(<16 x s32>) = G_CONCAT_VECTORS [[COPY1]](<8 x s32>), [[COPY]](<8 x s32>)
+    ; CHECK-NEXT: $x0 = COPY [[CONCAT_VECTORS]](<16 x s32>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $x0
+    %1:_(<8 x s32>) = COPY $wl2
+    %2:_(<8 x s32>) = COPY $wl4
+    %0:_(<16 x s32>) = G_SHUFFLE_VECTOR %1:_(<8 x s32>), %2:_, shufflemask(8, 9, 10, 11, 12, 13, 14, 15, 0, 1, 2, 3, 4, 5, 6, 7)
```

Comment:
Do we ever see this pattern? I'm curious, because it looks to me like a generic canonicalization to just swap the two inputs and rewrite the shufflemask to run from 0-15.
Do you know whether that canonicalization exists for LLVM IR?

Reply:
As far as, I can see, the answer is no. There is some analysis done `InstructionSimplify.cpp`, but they don't cover this case. If you feed it just LLVM IR for a shuffle vector, it will also not reorder it. But actually, while looking at it now, this code can be dropped since one of the earlier changes during a rewrite made this case be covered under an earlier commit

Reply:
Could you please move the tests from this commit to that earlier change then and drop the code change?

Reply:
Yes, of course, I was just running the tests before I pushed it. Never can be too sure